File: androidutil/src/main/java/com/akingyin/base/utils/HtmlUtils.java
package com.akingyin.base.utils;

import android.text.Spanned;
import androidx.core.text.HtmlCompat;

public class HtmlUtils {

	public   static  final  String SPACE  = "&nbsp;";
	public static final String getRedHtml(String content) {
		return "<font color=#FF0000>" + content + "</font>";
	}

	public static final String getGreenHtml(String content) {
		return "<font color=#009966>" + content + "</font>";
	}

	public static final String getBlueHtml(String content) {
		return "<font color=#0000FF>" + content + "</font>";
	}

	public static final String getColorHtml(String content, String color) {
		return "<font color=" + color + ">" + content + "</font>";
	}

	public static final String getSmallColorHtml(String content, String color) {
		return "<font color=" + color + "><small>" + content + "</small></font>";
	}

	public static final String getReadPHtml(String content) {
		return "<p style=font-size:30px;color:red>" + content + "</p>";
	}

	public static final String getBigColorHtml(String content, String color) {
		return "<font color=" + color + "><big>" + content + "</big></font>";
	}

	public static final String getBigStrongColorHtml(String content, String color) {
		return "<font color=" + color + "><strong><big>" + content + "</big></strong></font>";
	}

	public static final String getBigRedHtml(String content) {
		 return "<font color=#FF0000><big>" + content + "</big></font>";
	}
	public static final String getBigStrongRedHtml(String content) {
		return "<font color=#FF0000><strong><big>" + content + "</big></strong></font>";
	}

	public  static  final   String   getTelNumberHtml(String  phoneNumber){
		return "<a href='tel:"+phoneNumber+"'>"+phoneNumber+"</a>";
	}

	public   static Spanned  getTextHtml(String content){
		return HtmlCompat.fromHtml(content,HtmlCompat.FROM_HTML_MODE_LEGACY);
	}
}

File: androidutil/src/main/java/com/akingyin/base/utils/HtmlUtils.kt
package com.akingyin.base.utils

import android.text.Spanned
import androidx.core.text.HtmlCompat
import java.lang.StringBuilder

object HtmlUtils {
    const val SPACE = "&nbsp;"
    fun getRedHtml(content: String): String {
        return "<font color=#FF0000>$content</font>"
    }

    fun getManySpace(number : Int ):String{
        val stringBuilder = StringBuilder()
        for (i in 0.. number){
           stringBuilder.append(SPACE)
        }
        return stringBuilder.toString()
    }

    fun getGreenHtml(content: String): String {
        return "<font color=#009966>$content</font>"
    }

    fun getBlueHtml(content: String): String {
        return "<font color=#0000FF>$content</font>"
    }

    fun getColorHtml(content: String, color: String): String {
        return "<font color=$color>$content</font>"
    }

    fun getSmallColorHtml(content: String, color: String): String {
        return "<font color=$color><small>$content</small></font>"
    }

    fun getReadPHtml(content: String): String {
        return "<p style=font-size:30px;color:red>$content</p>"
    }

    fun getBigColorHtml(content: String, color: String): String {
        return "<font color=$color><big>$content</big></font>"
    }

    fun getBigStrongColorHtml(content: String, color: String): String {
        return "<font color=$color><strong><big>$content</big></strong></font>"
    }

    fun getBigRedHtml(content: String): String {
        return "<font color=#FF0000><big>$content</big></font>"
    }

    fun getBigStrongRedHtml(content: String): String {
        return "<font color=#FF0000><strong><big>$content</big></strong></font>"
    }

    fun getTelNumberHtml(phoneNumber: String): String {
        return "<a href='tel:$phoneNumber'>$phoneNumber</a>"
    }

    fun getTextHtml(content: String): Spanned {
        return HtmlCompat.fromHtml(content, HtmlCompat.FROM_HTML_MODE_LEGACY)
    }
}